src/hooks/useBasePopoverReactionList/index.ts

```diff
@@ -0,0 +1,131 @@
+
+export default function useBasePopoverReactionList({emojiName, emojiReactions, accountID, reportActionID, preferredLocale}: BasePopoverReactionListHookProps) {
+    const [isPopoverVisible, setIsPopoverVisible] = useState(false);
+    const [cursorRelativePosition, setCursorRelativePosition] = useState({horizontal: 0, vertical: 0});
+    const [popoverAnchorPosition, setPopoverAnchorPosition] = useState({horizontal: 0, vertical: 0});
+    const reactionListRef = useRef<ReactionListAnchor>(null);
+
+    function getReactionInformation() {
+        const selectedReaction = emojiReactions?.[emojiName];
+
+        if (!selectedReaction) {
+            // If there is no reaction, we return default values
+            return {
+                emojiName: '',
+                reactionCount: 0,
+                emojiCodes: [],
+                hasUserReacted: false,
+                users: [],
+            };
+        }
+
+        const {emojiCodes, reactionCount, hasUserReacted, userAccountIDs} = EmojiUtils.getEmojiReactionDetails(emojiName, selectedReaction, accountID);
+
+        const users = PersonalDetailsUtils.getPersonalDetailsByIDs(userAccountIDs, accountID, true);
+        return {
+            emojiName,
+            emojiCodes,
+            reactionCount,
+            hasUserReacted,
+            users,
+        };
+    }
+
+    /**
+     * Get the BasePopoverReactionList anchor position
+     * We calculate the achor coordinates from measureInWindow async method
+     */
+    function getReactionListMeasuredLocation(): Promise<{x: number; y: number}> {
+        return new Promise((resolve) => {
+            const reactionListAnchor = reactionListRef.current;
+            if (reactionListAnchor && 'measureInWindow' in reactionListAnchor) {
+                reactionListAnchor.measureInWindow((x, y) => resolve({x, y}));
+            } else {
+                // If the anchor is not available or does not have the measureInWindow method, we return 0, 0
+                resolve({x: 0, y: 0});
+            }
+        });
+    }
+
+    /**
+     * Show the ReactionList modal popover.
+     *
+     * @param event - Object -  A press event.
+     * @param reactionListAnchor - Element - reactionListAnchor
+     */
+    const showReactionList: ShowReactionList = (event, reactionListAnchor) => {
+        // We get the cursor coordinates and the reactionListAnchor coordinates to calculate the popover position
+        const nativeEvent = (event as SyntheticEvent<ReactionListAnchor, MouseEvent>)?.nativeEvent || {};
+        reactionListRef.current = reactionListAnchor;
+        getReactionListMeasuredLocation().then(({x, y}) => {
+            setCursorRelativePosition({horizontal: nativeEvent.pageX - x, vertical: nativeEvent.pageY - y});
+            setPopoverAnchorPosition({
+                horizontal: nativeEvent.pageX,
+                vertical: nativeEvent.pageY,
```

Comment:
Looks like `nativeEvent` can be an empty object and accessing pageX and pageY properties leads to undefined?
And `nativeEvent.pageX - x` results into NaN, right?
Should we return early if nativeEvent is empty?

Reply:
Thanks for your review @MonilBhavsar 
I don't think there would be any scenarios in which `nativeEvent` would be empty and causing such issues. Also this code was working like this before and I just moved it to its own hoo, So I'm not sure if changing its behavior, even if considered as an in-scope change for this issue, wouldn't cause any regressions.
If you're sure that it won't cause any further issues, please let me know and I will add it.

Reply:
Yeah, It was like this before. So I just wanted to clarify

Reply:
Thanks for your clarification. So do you think should I add an early return or is it fine for now?

Reply:
It is fine, thanks!